### ISSUE TITLE
auro-radio: unable to change radio selection with iOS VoiceOver

### ISSUE THREAD

**Issue by geoffrich** [AlaskaAirlines/auro-radio#36]
## Describe the bug
When using VoiceOver on iOS, I am unable to change the radio button I have selected. Focus is quickly returned to the currently selected radio button.

## To Reproduce
1. Navigate to the radio group example on the doc site with VoiceOver
2. Select the first radio button by double-tapping
3. Attempt to swipe to the next radio button. Focus briefly moved, but is quickly moved back to the first radio button

See this video for an example: 

https://user-images.githubusercontent.com/4992896/120698622-231fea80-c464-11eb-9865-7523f1136073.mov

## Expected behavior
I can change the selected radio button on iOS with VoiceOver.

## Smartphone (please complete the following information):
 - Device: iPhone 8
 - OS: iOS 14.5.1
 - Browser: Safari & Firefox

## Additional context
This is a severe accessibility bug.

**Comment by geoffrich** [comment]
I think the issue is with the `handleFocusSelected` function. Does anyone know why this function is necessary? Removing it makes the VoiceOver issue go away, and I can't tell if any functionality was lost.

https://github.com/AlaskaAirlines/auro-radio/blob/f6b59c1902cc4760e9d5fbd1b073ce763d53dfeb/src/auro-radio-group.js#L143-L147

**Comment by blackfalcon** [comment]
@tsalright you wrote this... any insights?

**Comment by geoffrich** [comment]
Running locally, I verified that removing the focus handling does not cause any tests to fail and does not prevent keyboard navigation or interacting with the radio with the mouse (tested on Firefox/Windows). I need to do more screen reader testing, but it seems to fix the VoiceOver issue.

Unless we can think of a good reason to keep it in, I would like to open a PR tomorrow that removes the `focusSelected` listener to fix the high-priority a11y issue. We can revisit replacing the functionality if we later discover that this caused a regression in other behavior.

**Comment by tsalright** [comment]
I think it had something to do with modifying the underlying input tag. Things may have changed since this was built, but we had to manually trigger the focus for tab order and styling to work on all browsers. I don't remember the details anymore.

**Comment by steve-nickerson** [comment]
This is an example of the behavior inside ascom's change reservation view. @geoffrich  

https://user-images.githubusercontent.com/16307172/120841629-9424d800-c520-11eb-9011-e4c6c5d7dde3.mp4

**Comment by blackfalcon** [comment]
@geoffrich piggy-backing on what @tsalright said... when we built that, there was a lot of 'hacking' we needed to do, and either this function was never needed or it is no longer needed. But if you feel confident that there is no regression with removing this and it fixes your issue. Then I say... DO IT!
